packages/camlp5/camlp5.7.09/opam

```diff
@@ -0,0 +1,52 @@
+synopsis: "Preprocessor-pretty-printer of OCaml"
+description:
+"""
+Camlp5 is a preprocessor and pretty-printer for OCaml programs. It also provides parsing and printing tools.
+
+As a preprocessor, it allows to:
+
+extend the syntax of OCaml,
+redefine the whole syntax of the language.
+As a pretty printer, it allows to:
+
+display OCaml programs in an elegant way,
+convert from one syntax to another,
+check the results of syntax extensions.
+Camlp5 also provides some parsing and pretty printing tools:
+
+extensible grammars
+extensible printers
+stream parsers and lexers
+pretty print module
+It works as a shell command and can also be used in the OCaml toplevel.
+"""
+opam-version: "2.0"
+maintainer: "https://github.com/ocaml/opam-repository/issues"
+authors: ["Daniel de Rauglaudre"]
+homepage: "https://camlp5.github.io"
+license: "BSD-3-Clause"
+bug-reports: "https://github.com/camlp5/camlp5/issues"
+dev-repo: "git+https://github.com/camlp5/camlp5.git"
+doc: "https://camlp5.github.io/doc/html"
+
+depends: [
+  "ocaml"       { >= "4.02" & <= "4.10.0" }
```

Comment:
```suggestion
  "ocaml"       { >= "4.08" & <= "4.10.0" }
```

It fails with all `ocaml` varsion smaller than 4.08:

```
# ocamlc.opt versdep.cmo ploc.cmo plexing.cmo plexer.cmo fstream.cmo gramext.cmo grammar.cmo diff.cmo extfold.cmo extfun.cmo pretty.cmo pprintf.cmo eprinter.cmo stdpp.cmo token.cmo -a -o gramlib.cma
# make[3]: Leaving directory '/home/opam/.opam/4.02/.opam-switch/build/camlp5.7.09/ocaml_src/lib'
# set -e; cd ocaml_src/lib; /usr/bin/make promote; cd ../..
# make[3]: Entering directory '/home/opam/.opam/4.02/.opam-switch/build/camlp5.7.09/ocaml_src/lib'
# cp versdep.cmo ploc.cmo plexing.cmo plexer.cmo fstream.cmo gramext.cmo grammar.cmo diff.cmo extfold.cmo extfun.cmo pretty.cmo pprintf.cmo eprinter.cmo stdpp.cmo token.cmo versdep.cmi ploc.cmi plexing.cmi plexer.cmi fstream.cmi gramext.cmi grammar.cmi diff.cmi extfold.cmi extfun.cmi pretty.cmi pprintf.cmi eprinter.cmi stdpp.cmi token.cmi ../../boot/.
# make[3]: Leaving directory '/home/opam/.opam/4.02/.opam-switch/build/camlp5.7.09/ocaml_src/lib'
# make[2]: Leaving directory '/home/opam/.opam/4.02/.opam-switch/build/camlp5.7.09'
# /usr/bin/make compile_cold
# make[2]: Entering directory '/home/opam/.opam/4.02/.opam-switch/build/camlp5.7.09'
# cd ocaml_src; set -e; \
# for i in lib odyl main meta; do \
#   cd $i; /usr/bin/make all; cd ..; \
# done; cd ..
# make[3]: Entering directory '/home/opam/.opam/4.02/.opam-switch/build/camlp5.7.09/ocaml_src/lib'
# make[3]: Nothing to be done for 'all'.
# make[3]: Leaving directory '/home/opam/.opam/4.02/.opam-switch/build/camlp5.7.09/ocaml_src/lib'
# make[3]: Entering directory '/home/opam/.opam/4.02/.opam-switch/build/camlp5.7.09/ocaml_src/odyl'
# echo "let standard_library =" > odyl_config.ml
# echo "  try Sys.getenv \"CAMLP5LIB\" with" >> odyl_config.ml
# echo "    Not_found -> \"/home/opam/.opam/4.02/lib/camlp5\"" | \
# sed -e 's|\\|/|g' >> odyl_config.ml
# cat odyl_config.ml > odyl_config.ppo
# ocamlc.opt -warn-error A -c -impl odyl_config.ppo
# rm -f odyl_config.ppo
# cat odyl_main.mli > odyl_main.ppi
# ocamlc.opt -warn-error A -c -intf odyl_main.ppi
# rm -f odyl_main.ppi
# cat odyl_main.ml > odyl_main.ppo
# ocamlc.opt -warn-error A -c -impl odyl_main.ppo
# File "odyl_main.ml", line 61, characters 19-40:
# Error: This variant pattern is expected to have type Dynlink.error
#        The constructor Module_already_loaded does not belong to type Dynlink.error
# make[3]: *** [../../config/Makefile:23: odyl_main.cmo] Error 2
# make[3]: Leaving directory '/home/opam/.opam/4.02/.opam-switch/build/camlp5.7.09/ocaml_src/odyl'
# make[2]: *** [Makefile:168: compile_cold] Error 2
# make[2]: Leaving directory '/home/opam/.opam/4.02/.opam-switch/build/camlp5.7.09'
# make[1]: *** [Makefile:36: boot/camlp5] Error 2
# make[1]: Leaving directory '/home/opam/.opam/4.02/.opam-switch/build/camlp5.7.09'
# make: *** [Makefile:150: world.opt] Error 2
```

Reply:
Indeed, I wrote this matching without protection so this requires ocaml 4.08.
Maybe I'll first ask Daniel whether this is fine.
Am I right that opam is same to cope with this (i.e. it will automatically select an older version for older versions of ocaml)?

Reply:
Yes, if you add the constraint that I suggest opam solver will select the version that works with each specific compiler